### ISSUE TITLE
Screen not clearing when environment's background is in canvas mode

### ISSUE THREAD

**Issue by MightyPrinny** [godotengine/godot#34070]
**Godot version:**
3.2 e64a663c5

**OS/device including version:**
MXLinux 64-bits

**Issue description:**
The screen doesn't clear when the environment's background mode is set to Canvas and canvas max layer is less than 0.

![image](https://user-images.githubusercontent.com/18746022/70005447-756b0d00-1548-11ea-9294-26a653213da4.png)

**Steps to reproduce:**
Make a scene, add an environment, set the background mode to canvas and canvas max layer to -1.

**Comment by FeralBytes** [comment]
Does this also happen when the game is exported? For instance if you were to move the node in between frames does the same thing happen, or is this just a problem for the IDE?

**Comment by MightyPrinny** [comment]
yes, it happens in the editor and in game, I tried on android too.

**Comment by MightyPrinny** [comment]
it's only noticeable if the background is empty, if there were a background image you wouldn't see it. Also when I tried to open the editor

**Comment by fossegutten** [comment]
Can you test with this commit, and see if it fixes it? #36822
Or post a project to test with

**Comment by spindlebink** [comment]
I know this is an old issue, but it's still occurring and I was wondering if there's any update on it. It doesn't appear to happen in `master`---is there a way whatever has fixed it could be backported to v3?

**Comment by Calinou** [comment]
> It doesn't appear to happen in master---is there a way whatever has fixed it could be backported to v3?

The rendering backend was rewritten from scratch in `master`, so it's unlikely that a fix can be backported.
